src/clients/SpokePoolClient.ts

```diff
@@ -664,25 +679,29 @@ export class SpokePoolClient extends BaseAbstractClient {
     // is heavy as there is a fair bit of block number lookups that need to happen. Note this call REQUIRES that the
     // hubPoolClient is updated on the first before this call as this needed the the L1 token mapping to each L2 token.
     if (eventsToQuery.includes("FundsDeposited")) {
-      const allDeposits = [
-        ...(queryResults[eventsToQuery.indexOf("FundsDeposited")] as FundsDepositedEvent[]),
-        ...this.earlyDeposits,
-      ];
-      const { earlyDeposits = [], depositEvents = [] } = groupBy(allDeposits, (depositEvent) => {
-        if (this._isEarlyDeposit(depositEvent, currentTime)) {
-          const { args, transactionHash } = depositEvent;
-          this.logger.debug({
-            at: "SpokePoolClient#update",
-            message: "Deferring early deposit event.",
-            currentTime,
-            deposit: { args, transactionHash },
-          });
-          return "earlyDeposits";
-        } else {
-          return "depositEvents";
-        }
-      });
+      // Filter out any early v2 deposits (quoteTimestamp > HubPoolClient.currentTime). Early deposits are no longer a
+      // critical risk in v3, so don't worry about filtering those. This will reduce complexity in several places.
+      const { earlyDeposits = [], v2DepositEvents = [] } = groupBy(
+        [
+          ...this.earlyDeposits,
+          ...((queryResults[eventsToQuery.indexOf("FundsDeposited")] ?? []) as FundsDepositedEvent[]),
+        ],
+        (depositEvent) => (this._isEarlyDeposit(depositEvent, currentTime) ? "earlyDeposits" : "v2DepositEvents")
+      );
+      if (earlyDeposits.length > 0) {
```

Comment:
One minor change here - we now log once for the sum of any early deposits, rather than once per deposit. The same information is available in the new consolidated log. Ultimately this will be removed once the v2 deposit functions are flushed; I anticipcate that we probably won't be concerned about early deposits in a v3-only world.

Reply:
This is a great QoL improvement, the logs were really busy